- Description: dasd: DIAG access does not work with disks which have not been
               CMS reserved.
  Symptom:     Trying to access a disk via the DIAG discipline which has been
               formatted, but not CMS reserved, will fail.
  Problem:     There is an obsolete DASD-internal check for CMS reserved data
               when trying to access a DASD disk via the DIAG discipline.
  Solution:    Remove the obsolete check.
  Problem-ID:  -
  Archs:       s390-31

--- linux-2.5/drivers/s390/block/dasd_diag.c	25 May 2004 13:38:00 -0000	1.34.2.1
+++ linux-2.5/drivers/s390/block/dasd_diag.c	28 May 2004 12:17:04 -0000	1.34.2.2
@@ -6,7 +6,7 @@
  * Bugreports.to..: <Linux390@de.ibm.com>
  * (C) IBM Corporation, IBM Deutschland Entwicklung GmbH, 1999,2000
  *
- * $Revision: 1.34.2.1 $
+ * $Revision: 1.34.2.2 $
  */
 
 #include <linux/config.h>
@@ -294,7 +294,7 @@
 		mdsk_term_io(device);
 	}
 	if (bsize <= PAGE_SIZE && label[3] == bsize &&
-	    label[0] == 0xc3d4e2f1 && label[13] != 0) {
+	    label[0] == 0xc3d4e2f1) {
 		device->blocks = label[7];
 		device->bp_block = bsize;
 		device->s2b_shift = 0;	/* bits to shift 512 to get a block */

